be/src/runtime/runtime_query_statistics_mgr.cpp

```diff
@@ -17,14 +17,432 @@
+                    "details.",
+                    PrintThriftNetworkAddress(coor_addr), e.what());
+        }
+        return Status::RpcError("Send stats failed");
+    } catch (std::exception& e) {
+        LOG_WARNING("Failed to send statistics to {}, reason: {}, you can see fe log for details.",
+                    PrintThriftNetworkAddress(coor_addr), e.what());
+        return Status::RpcError("Send stats failed");
+    }
+
+    return Status::OK();
+}
+
+TReportExecStatusParams RuntimeQueryStatiticsMgr::create_report_exec_status_params_x(
+        const TUniqueId& query_id,
+        const std::unordered_map<int32, std::vector<std::shared_ptr<TRuntimeProfileTree>>>&
+                fragment_id_to_profile,
+        const std::vector<std::shared_ptr<TRuntimeProfileTree>>& load_channel_profiles) {
+    TQueryProfile profile;
+    profile.__set_query_id(query_id);
+
+    std::map<int32_t, std::vector<TDetailedReportParams>> fragment_id_to_profile_req;
+
+    for (const auto& entry : fragment_id_to_profile) {
+        int32_t fragment_id = entry.first;
+        const std::vector<std::shared_ptr<TRuntimeProfileTree>>& fragment_profile = entry.second;
+        std::vector<TDetailedReportParams> detailed_params;
+
+        for (auto pipeline_profile : fragment_profile) {
+            if (pipeline_profile == nullptr) {
+                auto msg = fmt::format("Register fragment profile {} {} failed, profile is null",
+                                       print_id(query_id), fragment_id);
+                DCHECK(false) << msg;
+                LOG_ERROR(msg);
+                continue;
+            }
+
+            TDetailedReportParams tmp;
+            tmp.__set_profile(*pipeline_profile);
+            // tmp.fragment_instance_id is not needed for pipeline x
+            detailed_params.push_back(tmp);
+        }
+
+        fragment_id_to_profile_req.insert(std::make_pair(fragment_id, detailed_params));
+    }
+
+    if (fragment_id_to_profile_req.size() == 0) {
+        LOG_WARNING("No fragment profile found for query {}", print_id(query_id));
+    }
+
+    profile.__set_fragment_id_to_profile(fragment_id_to_profile_req);
+
+    std::vector<TRuntimeProfileTree> load_channel_profiles_req;
+    for (auto load_channel_profile : load_channel_profiles) {
+        if (load_channel_profile == nullptr) {
+            auto msg = fmt::format(
+                    "Register fragment profile {} {} failed, load channel profile is null",
+                    print_id(query_id), -1);
+            DCHECK(false) << msg;
+            LOG_ERROR(msg);
+            continue;
+        }
+
+        load_channel_profiles_req.push_back(*load_channel_profile);
+    }
+
+    if (load_channel_profiles_req.size() > 0) {
+        profile.__set_load_channel_profiles(load_channel_profiles_req);
+    }
+
+    TReportExecStatusParams req;
+    req.__set_query_profile(profile);
+    req.__set_query_id(TUniqueId());
+
```

Comment:
Should also set backend_id because FE need backendid to dedup profiles.

Reply:
done